feature/google/src/test/java/com/revenuecat/purchases/google/BillingWrapperTest.kt

```diff
@@ -1505,6 +1505,59 @@ class BillingWrapperTest {
         }
     }
 
+    @Test
+    fun `querySkuDetailsAsync only calls one response when BillingClient responds twice`() {
+        var numCallbacks = 0
+
+        val slot = slot<SkuDetailsResponseListener>()
+        every {
+            mockClient.querySkuDetailsAsync(
+                any(),
+                capture(slot)
+            )
+        } answers {
+            slot.captured.onSkuDetailsResponse(billingClientOKResult, null)
+            slot.captured.onSkuDetailsResponse(billingClientOKResult, null)
+        }
+
+        wrapper.querySkuDetailsAsync(
+            ProductType.SUBS,
+            setOf("", ""),
+            {
+                numCallbacks++
+            }, {
+                numCallbacks++
+            })
+
+        assertThat(numCallbacks == 1)
+    }
+
+    @Test
+    fun `queryPurchaseHistoryAsync only calls one response when BillingClient responds twice`() {
+        var numCallbacks = 0
+
+        val slot = slot<PurchaseHistoryResponseListener>()
+        every {
+            mockClient.queryPurchaseHistoryAsync(
+                any(),
+                capture(slot)
+            )
+        } answers {
+            slot.captured.onPurchaseHistoryResponse(billingClientOKResult, null)
+            slot.captured.onPurchaseHistoryResponse(billingClientOKResult, null)
```

Comment:
this covers the case where completion is called more than once, serially. 
In my other comment I mentioned that I believe there can still be issues if completion gets called more than once, but concurrently.
It's hard to make a test for this, though. 

I was thinking, if we had a test that: 
- calls `wrapper.queryPurchaseHistoryAsync`, with a completion that takes long (like with a `sleep`)
- has the `slot.captured.onPurchaseHistoryResponse(billingClientOKResult, null)` bit called twice, but from different threads, so it's not serial and doesn't wait for the `hasResponded = true` line

We might be able to have a test that consistently reproduces the case I mentioned and ensures we cover it